File: R-example/tests/testthat/test-example.R
# A test for the example script.

test_that("mysum works", {
  expect_equal(mysum(0, 0), 0)
  expect_equal(mysum(1, 0), 1)
  expect_equal(mysum(0, 1), 1)
  expect_equal(mysum(1, 1), 2)

  expect_equal(mysum(-1, 0), -1)
  expect_equal(mysum(0, -1), -1)
  expect_equal(mysum(-1, -1), -2)

  expect_equal(mysum(-1, 1), 0)
  expect_equal(mysum(1, -1), 0)
})

File: python/test_example.py
"""
A test for the example script.
"""

from example import mysum


def test_mysum():
    assert mysum(0, 0) == 0
    assert mysum(1, 0) == 1
    assert mysum(0, 1) == 1
    assert mysum(1, 1) == 2

    assert mysum(-1, 0) == -1
    assert mysum(0, -1) == -1
    assert mysum(-1, -1) == -2

    assert mysum(-1, 1) == 0
    assert mysum(1, -1) == 0
